redash/tasks/queries.py

```diff
@@ -190,12 +191,18 @@ def refresh_queries():
             elif query.data_source.paused:
                 logging.info("Skipping refresh of %s because datasource - %s is paused (%s).", query.id, query.data_source.name, query.data_source.pause_reason)
             else:
-                if query.options and len(query.options.get('parameters', [])) > 0:
-                    query_params = {p['name']: p.get('value')
-                                    for p in query.options['parameters']}
-                    query_text = query.parameterized.apply(query_params).query
-                else:
-                    query_text = query.query_text
+                query_text = query.query_text
+
+                parameters = {p['name']: p.get('value') for p in query.options.get('parameters', [])}
+                if any(parameters):
+                    try:
+                        query_text = query.parameterized.apply(parameters).query
+                    except InvalidParameterError as e:
```

Comment:
I tend to think these should actually not be skipped but rather executed (with an unparameterized `query_text`) and fail - so they will show up in failure reports, in order to give users the chance to rectify them.

Reply:
We do the reporting at a different stage and hence why we can't report on them. If we move the parameter merging to the query task, then we can report on this issue.

Reply:
Heh, I was actually thinking of a far less elegant solution - when unable to do parameter merging - just let a broken unapplied query run and have the DB report a failure, which would end up in the report :)

Reply:
One caveat about moving the parameter merging to the query task - `enqueue_query` creates job locks according to the hashed query. If we don't apply the parameters before calling `enqueue_query`, we will lock according to hashed versions of strings like `SELECT {{foo}}`, which could prevent different queries from running if they hash to the same value.

This is probably a non-issue because scheduled parameterized queries use their default values, so they hash to the same value regardless. It's just disturbing to see.

Reply:
We should revise the hashing logic anyway, as it's currently lower cases the query. We can hash the query text + parameter values to de-duplicate based on actual query without merging in the values.

Reply:
Another option is to report from here directly to the error report. We should also increment the schedule failures counter to avoid having to reprocess these queries all the time.

Out of scope for this PR, but worth keeping in mind.

Reply:
Went ahead with the latter. Let me know if you want to keep cb82a53 or discard it.